common/djangoapps/django_sudo_helpers/pipeline.py

```diff
@@ -0,0 +1,38 @@
+"""django-sudo pipeline definitions."""
+
+import re
+
+from django.conf import settings
+from social.pipeline import partial
+
+from sudo.utils import grant_sudo_privileges
+from util.request import COURSE_REGEX
+
+AUTH_ENTRY_SUDO = 'grant_sudo'
+
+# pylint: disable=unused-argument
+
+
+@partial.partial
+def create_sudo_session(strategy, auth_entry, *args, **kwargs):
+    """
+    Grant sudo privileges if user is authenticated and auth_entry
+    is equal to AUTH_ENTRY_SUDO.
+    """
+    request = strategy.request if strategy else None
+    if auth_entry == AUTH_ENTRY_SUDO and request.user.is_authenticated():
+        _region = None
+        redirect_to = strategy.session_get('next', '')
+        match = COURSE_REGEX.match(redirect_to)
+        if not match:
+            # COURSE_REGEX will not work for studio
+            course_regex = re.compile(r'^.*?/course_team/{}'.format(settings.COURSE_ID_PATTERN))
```

Comment:
I see. If we want to be certain that we're matching against valid course_ids, we could match the `COURSE_REGEX`/`COURSE_ID_PATTERN` against the URL, and then try to load each match using `CourseKey.from_string()`. Whichever one is valid as a course key would be the region. Then we don't have to special-case certain URLs, which will be difficult to maintain.

The performance cost of that approach might not be great, but since this method is unlikely to be called often, I don't think we'd have to worry about it... @cpennington does this concern you?